### PR TITLE
Fix dotnet test Command Not Recognizing NUnit3TestAdapter

### PR DESCRIPTION
**Pull request by mobilebilly**
Fix for Issue #109

This fix addresses the issue where the dotnet test command cannot recognize the NUnit3TestAdapter in existing ItemGroup elements.

Changes Made:

Moved the reference to NUnit3TestAdapter to an ItemGroup without a TargetFramework condition.

### CODE REVIEW

**Comment by FreeAndNil** [comment]
@mobilebilly Thank you very much for your contribution.
I retargeted you pull request to Feature/111... because this is our current development branch for 3.0.0.
Can you please fix the conflicts and test the changes?

**Comment by mobilebilly** [comment]
@FreeAndNil , no problem.
The conflicts have been fixed, and my branch is now up-to-date with the latest changes from the upstream branch Feature/111.... 
Please review the updated pull request and let me know if there are any further changes needed.

File: src/log4net.Tests/log4net.Tests.csproj
Comment (FreeAndNil) [inline src/log4net.Tests/log4net.Tests.csproj:1]: Can you please change lines 2 to 48 to the following?
```
  <PropertyGroup>
    <IsTestProject>true</IsTestProject>
    <TargetFrameworks>net462;net8.0</TargetFrameworks>
    <NoWarn>NETSDK1138;CS1701</NoWarn>
    <OutputType>Library</OutputType>
    <OutputPath>bin\$(Configuration)</OutputPath>
    <Configurations>Debug;Release</Configurations>
    <Platforms>AnyCPU</Platforms>
    <Deterministic>true</Deterministic>
    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
    <DefineConstants>TRACE;DEBUG;$(DefineConstants)</DefineConstants>
    <!-- suppress analyzer mismatch warning -->
    <NoWarn>CS8032</NoWarn>
  </PropertyGroup>
```
The line `<IsTestProject>true</IsTestProject>` is really important, without it only net8.0 is executed and net462 is skipped.
The other lines are just removing no longer needed garbage from the old style project file.

**Review by FreeAndNil — CHANGES_REQUESTED** [review]
Please adjust as proposed, then we can merge.

**Comment by mobilebilly** [comment]
@FreeAndNil, the additional change in the test project is done. thank you

**Review by FreeAndNil — APPROVED** [review]
Thanks a lot.

**Pull request merged**